### PR TITLE
Source code CDEvents to render as CloudEvent and schema validation

### PR DESCRIPTION
**Pull request by rjalander**
This PR is a continuation of https://github.com/cdevents/sdk-java/pull/35, to implement other Source code version control events

### CODE REVIEW

File: src/main/java/dev/cdevents/events/ChangeAbandonedCDEvent.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import dev.cdevents.models.BranchSubject;
+import dev.cdevents.models.CDEvent;
+
+import java.net.URI;
+
+public class ChangeAbandonedCDEvent extends CDEvent {
+
+    private static final String CDEVENT_VERSION = "0.1.0";
+    @JsonProperty(required = true)
+    private BranchSubject subject;
```
[the remark this answers is not in the document]
Comment (rjalander) [reply src/main/java/dev/cdevents/events/ChangeAbandonedCDEvent.java:14]: Actually, branch and change events have similar properties for all the event types `id` `source` and `repository` as per spec. So used the same model for Change and branch events. I can rename the class file name to apply this as a common model

[the remark this answers is not in the document]
Comment (afrittoli) [reply src/main/java/dev/cdevents/events/ChangeAbandonedCDEvent.java:14]: They have today but they may diverge in future, there's nothing in the spec that enforces them to be identical and it seems confusing that change events should rely on the branch model

[the remark this answers is not in the document]
Comment (rjalander) [reply src/main/java/dev/cdevents/events/ChangeAbandonedCDEvent.java]: Yeah, I got that. Created `ChangeSubject` model to separate them.

**Review by afrittoli — APPROVED** [review]
Thanks @rjalander - this looks good.

**Comment by afrittoli** [comment]
@rjalander - you will need to solve merged conflicts before this can be merged

**Pull request merged**